libs/utils/android/benchmark.py

```diff
@@ -0,0 +1,287 @@
+        LisaLogging.setup()
+        self._log = logging.getLogger('Benchmark')
+
+        self._log.info('=== CommandLine parsing...')
+        self.args = self._parseCommandLine()
+
+        self._log.info('=== TestEnv setup...')
+        self.bm_conf = self._getBmConf()
+        self.te = TestEnv(self.bm_conf)
+        self.target = self.te.target
+
+        self._log.info('=== Initialization...')
+        self.wl = self._getBmName()
+        self.out_dir=self.te.res_dir
+        try:
+            self.benchmarkInit()
+        except:
+            self._log.warning('Benchmark initialization failed: execution aborted')
+            raise
+
+        self._log.info('=== Execution...')
+        self.wl.run(out_dir=self.out_dir,
+                    collect=self._getBmCollect(),
+                    **self.bm_params)
+
+        self._log.info('=== Finalization...')
+        self.benchmarkFinalize()
+
+    def _wait_for_logcat_idle(self, seconds=1):
+        lines = 0
+
+        # Clear logcat
+        # os.system('{} logcat -s {} -c'.format(adb, DEVICE));
+        self.target.clear_logcat()
+
+        # Dump logcat output
+        logcat_cmd = 'adb -s {} logcat'.format(self.target.adb_name)
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+        logcat_poll = select.poll()
+        logcat_poll.register(logcat.stdout, select.POLLIN)
+
+        # Monitor logcat until it's idle for the specified number of [s]
+        self._log.info('Waiting for system to be almost idle')
+        self._log.info('   i.e. at least %d[s] of no logcat messages', seconds)
+        while True:
+            poll_result = logcat_poll.poll(seconds * 1000)
+            if not poll_result:
+                break
+            lines = lines + 1
+            line = logcat.stdout.readline(1024)
+            if lines % 1000:
+                self._log.info('   still waiting...')
+            if lines > 1e6:
+                self._log.warning('device logcat seems quite busy, '
+                                  'continuing anyway... ')
+
+    def reboot_target(self, disable_charge=True):
+        """
+        Reboot the target if a "boot-image" has been specified
+
+        If the user specify a boot-image as a command line parameter, this
+        method will reboot the target with the specified kernel and wait
+        for the target to be up and running.
+        """
+
+        # Reboot the device, if a boot_image has been specified
+        if self.args.boot_image:
+
+            self._log.warning('=== Rebooting...')
+            self._log.warning('Rebooting image to use: %s', self.args.boot_image)
+
+            self._log.debug('Waiting 6[s] to enter bootloader...')
+            self.target.adb_reboot_bootloader('reboot-bootloader')
+            sleep(6)
+            # self._fastboot('boot {}'.format(self.args.boot_image))
+            cmd = 'boot {}'.format(self.args.boot_image)
+            self.target.fastboot_command(cmd, device=self.target.adb_name)
```

Comment:
That has been a blind rename of the originally locally defined fastboot_command... which was also taking care to use the fastboot command provided by the ANDROID_HOME configuration.

Devlib, once again, is a bit naive in that sense... and just assume that the system provide fastboot command is working. We had issues in the past, using un-released AOSP SDK versions, because of that...  thus, I guess I'll probably better rollback to the local command while we devlib will be fixed to allow user-defined fastboot binaries.